File: backend/src/main/java/tom/projects/monkeylog/model/measurement/MeasurementPoint.java
package tom.projects.monkeylog.model.measurement;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UuidGenerator;
import tom.projects.monkeylog.model.user.UserOwned;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class MeasurementPoint implements UserOwned {
    @Id
    @GeneratedValue
    @UuidGenerator
    private UUID id;

    private Double value;

    @ManyToOne
    @JoinColumn(name = "measurement_id", nullable = false)
    private Measurement measurement;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @Override
    public UUID getUserId() {
        return getMeasurement().getUserId();
    }
}

File: backend/bin/main/tom/com/monkeylog/model/measurement/MeasurementPoint.kt
package tom.com.monkeylog.model.measurement

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UuidGenerator
import tom.com.monkeylog.model.user.UserOwned
import java.time.LocalDateTime
import java.util.*

@Entity
class MeasurementPoint(
    @Id
    @GeneratedValue
    @UuidGenerator
    var id: UUID? = null,
    var value: Double,
    @ManyToOne
    @JoinColumn(name = "measurement_id", nullable = false)
    var measurement: Measurement,
    @CreationTimestamp
    var createdAt: LocalDateTime? = null,
    override var userId: UUID? = null
) : UserOwned